tests/tracer/telemetry/test_writer.py

```diff
@@ -38,6 +38,8 @@ def mock_time():
 @pytest.fixture
 def telemetry_writer():
     telemetry_writer = TelemetryWriter(AGENT_URL)
+    # Enable the TelemetryWriter without queuing an app-started event
+    # and setting up exit hooks
     telemetry_writer._enabled = True
```

Comment:
This fixture masked the regression introduced in an earlier PR. I left a comment here to explain why setting `telemetry_writer._enabled = True` is necessary. Instead of leaving this comment I could try to refactor the tests instead.